File: TestProject1/Class/UnitTest_ClsCmdStatus.cs
using CmnClsLib.Module;
using CmnClsLib.Class;
using CmnClsLib.Interface;
using System.Text;
using System.Linq;
using Xunit;
using Assert = Xunit.Assert;

namespace TestProject1.Class
{
    public class UnitTest_ClsCmdStatus
    {
        private static ClsLogger _logger = new();
        private ClsCmdStatus _objCmdStatus = new(_logger);

        // --------------------------------------------------------------------
        // public void Initialize()
        // --------------------------------------------------------------------
        [Fact]
        public void カンマ区切り文字列に値を設定してInitを実行するとそれぞれの値がList型に分解されること()
        {
            _objCmdStatus.OkReturnCodeCsv = "0,5,10";
            _objCmdStatus.WarnReturnCodeCsv = "0,5,10";
            _objCmdStatus.ErrorReturnCodeCsv = "0,5,10";
            _objCmdStatus.OkMessageCsv = "正常,いけてる,完璧";
            _objCmdStatus.WarnMessageCsv = "警告,いけてるないかも,完璧でなかった";
            _objCmdStatus.ErrorMessageCsv = "異常,いけてない,失敗";
            _objCmdStatus.Initialize();
            Assert.Equal(3, _objCmdStatus.OkReturnCodeList.Count);
            Assert.Contains(0, _objCmdStatus.OkReturnCodeList);
            Assert.Contains(5, _objCmdStatus.OkReturnCodeList);
            Assert.Contains(10, _objCmdStatus.OkReturnCodeList);
            Assert.Equal(3, _objCmdStatus.WarnReturnCodeList.Count);
            Assert.Contains(0, _objCmdStatus.WarnReturnCodeList);
            Assert.Contains(5, _objCmdStatus.WarnReturnCodeList);
            Assert.Contains(10, _objCmdStatus.WarnReturnCodeList);
            Assert.Equal(3, _objCmdStatus.ErrorReturnCodeList.Count);
            Assert.Contains(0, _objCmdStatus.ErrorReturnCodeList);
            Assert.Contains(5, _objCmdStatus.ErrorReturnCodeList);
            Assert.Contains(10, _objCmdStatus.ErrorReturnCodeList);
            Assert.Equal(3, _objCmdStatus.OkMessageList.Count);
            Assert.Contains("正常", _objCmdStatus.OkMessageList);
            Assert.Contains("いけてる", _objCmdStatus.OkMessageList);
            Assert.Contains("完璧", _objCmdStatus.OkMessageList);
            Assert.Equal(3, _objCmdStatus.WarnMessageList.Count);
            Assert.Contains("警告", _objCmdStatus.WarnMessageList);
            Assert.Contains("いけてるないかも", _objCmdStatus.WarnMessageList);
            Assert.Contains("完璧でなかった", _objCmdStatus.WarnMessageList);
            Assert.Equal(3, _objCmdStatus.ErrorMessageList.Count);
            Assert.Contains("異常", _objCmdStatus.ErrorMessageList);
            Assert.Contains("いけてない", _objCmdStatus.ErrorMessageList);
            Assert.Contains("失敗", _objCmdStatus.ErrorMessageList);
        }

        // --------------------------------------------------------------------
        // public Boolean ShouldCheckMessage()
        // --------------------------------------------------------------------
        [Fact]
        public void チェックワードリストが空の場合はfalseを返却すること()
        {
            _objCmdStatus.OkMessageCsv = "";
            _objCmdStatus.WarnMessageCsv = "";
            _objCmdStatus.ErrorMessageCsv = "";
            _objCmdStatus.Initialize();
            Assert.False(_objCmdStatus.ShouldCheckMessage());
        }
        [Fact]
        public void チェックワードリストのどれかが空でない場合はtrueを返却すること()
        {
            _objCmdStatus.OkMessageCsv = "";
            _objCmdStatus.WarnMessageCsv = "警告,いけてるないかも,完璧でなかった";
            _objCmdStatus.ErrorMessageCsv = "";
            _objCmdStatus.Initialize();
            Assert.True(_objCmdStatus.ShouldCheckMessage());
        }

        // --------------------------------------------------------------------
        // public Boolean InitFlags()
        // --------------------------------------------------------------------
        [Fact]
        public void InitFlagsを実行するとフラグは全てfalseを返却すること()
        {
            _objCmdStatus.IsOkMessageHit = true;
            _objCmdStatus.IsWarnMessageHit = true;
            _objCmdStatus.IsErrorMessageHit = true;
            _objCmdStatus.ResetFlags();
            Assert.False(_objCmdStatus.IsOkMessageHit);
            Assert.False(_objCmdStatus.IsWarnMessageHit);
            Assert.False(_objCmdStatus.IsErrorMessageHit);
        }

        // --------------------------------------------------------------------
        // public void CheckMessageLine(String strLine)
        // --------------------------------------------------------------------
        [Fact]
        public void CheckMessageLineで正常終了しましたを評価すると正常該当フラグだけがtrueとなること()
        {
            _objCmdStatus.OkMessageCsv = "正常,いけてる,完璧";
            _objCmdStatus.WarnMessageCsv = "警告,いけてるないかも,完璧でなかった";
            _objCmdStatus.ErrorMessageCsv = "異常,いけてない,失敗";
            _objCmdStatus.Initialize();
            _objCmdStatus.ResetFlags();
            _objCmdStatus.CheckMessageLine("正常終了しました");
            Assert.True(_objCmdStatus.IsOkMessageHit);
            Assert.False(_objCmdStatus.IsWarnMessageHit);
            Assert.False(_objCmdStatus.IsErrorMessageHit);
        }
        [Fact]
        public void CheckMessageLineで警告が発生しましたがなんとか終了しましたを評価すると警告該当フラグだけがtrueとなること()
        {
            _objCmdStatus.OkMessageCsv = "正常,いけてる,完璧";
            _objCmdStatus.WarnMessageCsv = "警告,いけてるないかも,完璧でなかった";
            _objCmdStatus.ErrorMessageCsv = "異常,いけてない,失敗";
            _objCmdStatus.Initialize();
            _objCmdStatus.ResetFlags();
            _objCmdStatus.CheckMessageLine("警告が発生しましたがなんとか終了しました");
            Assert.False(_objCmdStatus.IsOkMessageHit);
            Assert.True(_objCmdStatus.IsWarnMessageHit);
            Assert.False(_objCmdStatus.IsErrorMessageHit);
        }
        [Fact]
        public void CheckMessageLineで異常終了しましたを評価すると異常該当フラグだけがtrueとなること()
        {
            _objCmdStatus.OkMessageCsv = "正常,いけてる,完璧";
            _objCmdStatus.WarnMessageCsv = "警告,いけてるないかも,完璧でなかった";
            _objCmdStatus.ErrorMessageCsv = "異常,いけてない,失敗";
            _objCmdStatus.Initialize();
            _objCmdStatus.ResetFlags();
            _objCmdStatus.CheckMessageLine("異常終了しました");
            Assert.False(_objCmdStatus.IsOkMessageHit);
            Assert.False(_objCmdStatus.IsWarnMessageHit);
            Assert.True(_objCmdStatus.IsErrorMessageHit);
        }
        [Fact]
        public void CheckMessageLineで警告が発生しましたが正常に終了しましたを評価すると正常と警告該当フラグがtrueとなること()
        {
            _objCmdStatus.OkMessageCsv = "正常,いけてる,完璧";
            _objCmdStatus.WarnMessageCsv = "警告,いけてるないかも,完璧でなかった";
            _objCmdStatus.ErrorMessageCsv = "異常,いけてない,失敗";
            _objCmdStatus.Initialize();
            _objCmdStatus.ResetFlags();
            _objCmdStatus.CheckMessageLine("警告が発生しましたが正常に終了しました");
            Assert.True(_objCmdStatus.IsOkMessageHit);
            Assert.True(_objCmdStatus.IsWarnMessageHit);
            Assert.False(_objCmdStatus.IsErrorMessageHit);
        }

        // --------------------------------------------------------------------
        // public void CheckCmdExitCode(int intCmdExitCode)
        // --------------------------------------------------------------------
        [Fact]
        public void BlnIsAlwaysNormalが正の場合はCheckCmdExitCodeの評価で結果がLVL_Iになること()
        {
            _objCmdStatus.IsAlwaysNormal = true;                                // 常に正常終了フラグON
            _objCmdStatus.CheckCommandExitCode(9999);
            Assert.Equal(MdlConst.LVL_I, _objCmdStatus.MethodExitStatus);    // 0
            Assert.Equal(MdlConst.LVL_I, _objCmdStatus.ReturnLevel);         // 正常
        }
        [Fact]
        public void 警告閾値と異常閾値が設定されていない場合にCheckCmdExitCodeの評価で0の結果がLVL_Iになること()
        {
            int intCmdRetcode = 0;                                              // コマンド戻り値
            _objCmdStatus.WarnThreshold = MdlConst.INT_NULL;                    // 未設定
            _objCmdStatus.ErrorThreshold = MdlConst.INT_NULL;                   // 未設定
            _objCmdStatus.CheckCommandExitCode(intCmdRetcode);
            Assert.Equal(MdlConst.LVL_I, _objCmdStatus.MethodExitStatus);    // 0
            Assert.Equal(MdlConst.LVL_I, _objCmdStatus.ReturnLevel);         // 正常
        }
        [Fact]
        public void 警告閾値と異常閾値が設定されていない場合にCheckCmdExitCodeの評価で1の結果レベルがLVL_Eになること()
        {
            int intCmdRetcode = 1;                                              // コマンド戻り値
            _objCmdStatus.WarnThreshold = MdlConst.INT_NULL;                    // 未設定
            _objCmdStatus.ErrorThreshold = MdlConst.INT_NULL;                   // 未設定
            _objCmdStatus.CheckCommandExitCode(intCmdRetcode);
            Assert.Equal(intCmdRetcode, _objCmdStatus.MethodExitStatus);     // コマンドの戻値
            Assert.Equal(MdlConst.LVL_E, _objCmdStatus.ReturnLevel);         // 異常
        }
        [Fact]
        public void 警告閾値と異常閾値が設定されていない場合にCheckCmdExitCodeの評価で異常時終了コードが20の場合1の戻り値は20になること()
        {
            int intCmdRetcode = 1;                                              // コマンド戻り値
            _objCmdStatus.WarnThreshold = MdlConst.INT_NULL;                    // 未設定
            _objCmdStatus.ErrorThreshold = MdlConst.INT_NULL;                   // 未設定
            _objCmdStatus.ErrorCode = 20;                                       // 異常時終了コード
            _objCmdStatus.CheckCommandExitCode(intCmdRetcode);
            Assert.Equal(20, _objCmdStatus.MethodExitStatus);                // 指定された異常時終了コード
            Assert.Equal(MdlConst.LVL_E, _objCmdStatus.ReturnLevel);         // 異常
        }

    }
}

File: src/test/java/tool/cmnclslib/cls/UnitTest_ClsCmdStatus.java
package tool.cmnclslib.cls;

import org.junit.Before;
import org.junit.Test;
import tool.cmnclslib.mdl.MdlConst;
import static org.junit.Assert.*;

/**
 * ClsCmdStatus の単体テストクラスです。
 */
public class UnitTest_ClsCmdStatus {

    private ClsLogger logger = new ClsLogger();
    private ClsCmdStatus objCmdStatus;

    @Before
    public void setUp() {
        objCmdStatus = new ClsCmdStatus(logger);
    }

    @Test
    public void カンマ区切り文字列に値を設定してInitを実行するとそれぞれの値がList型に分解されること() {
        objCmdStatus.setOkReturnCodeCsv("0,5,10");
        objCmdStatus.setWarnReturnCodeCsv("0,5,10");
        objCmdStatus.setErrorReturnCodeCsv("0,5,10");
        objCmdStatus.setOkMessageCsv("正常,いけてる,完璧");
        objCmdStatus.setWarnMessageCsv("警告,いけてるないかも,完璧でなかった");
        objCmdStatus.setErrorMessageCsv("異常,いけてない,失敗");
        objCmdStatus.initialize();

        assertEquals(3, objCmdStatus.getOkReturnCodeList().size());
        assertTrue(objCmdStatus.getOkReturnCodeList().contains(0));
        assertTrue(objCmdStatus.getOkReturnCodeList().contains(5));
        assertTrue(objCmdStatus.getOkReturnCodeList().contains(10));

        assertEquals(3, objCmdStatus.getWarnReturnCodeList().size());
        assertTrue(objCmdStatus.getWarnReturnCodeList().contains(0));
        assertTrue(objCmdStatus.getWarnReturnCodeList().contains(5));
        assertTrue(objCmdStatus.getWarnReturnCodeList().contains(10));

        assertEquals(3, objCmdStatus.getErrorReturnCodeList().size());
        assertTrue(objCmdStatus.getErrorReturnCodeList().contains(0));
        assertTrue(objCmdStatus.getErrorReturnCodeList().contains(5));
        assertTrue(objCmdStatus.getErrorReturnCodeList().contains(10));

        assertEquals(3, objCmdStatus.getOkMessageList().size());
        assertTrue(objCmdStatus.getOkMessageList().contains("正常"));
        assertTrue(objCmdStatus.getOkMessageList().contains("いけてる"));
        assertTrue(objCmdStatus.getOkMessageList().contains("完璧"));

        assertEquals(3, objCmdStatus.getWarnMessageList().size());
        assertTrue(objCmdStatus.getWarnMessageList().contains("警告"));
        assertTrue(objCmdStatus.getWarnMessageList().contains("いけてるないかも"));
        assertTrue(objCmdStatus.getWarnMessageList().contains("完璧でなかった"));

        assertEquals(3, objCmdStatus.getErrorMessageList().size());
        assertTrue(objCmdStatus.getErrorMessageList().contains("異常"));
        assertTrue(objCmdStatus.getErrorMessageList().contains("いけてない"));
        assertTrue(objCmdStatus.getErrorMessageList().contains("失敗"));
    }

    @Test
    public void チェックワードリストが空の場合はfalseを返却すること() {
        objCmdStatus.setOkMessageCsv("");
        objCmdStatus.setWarnMessageCsv("");
        objCmdStatus.setErrorMessageCsv("");
        objCmdStatus.initialize();
        assertFalse(objCmdStatus.shouldCheckMessage());
    }

    @Test
    public void チェックワードリストのどれかが空でない場合はtrueを返却すること() {
        objCmdStatus.setOkMessageCsv("");
        objCmdStatus.setWarnMessageCsv("警告,いけてるないかも,完璧でなかった");
        objCmdStatus.setErrorMessageCsv("");
        objCmdStatus.initialize();
        assertTrue(objCmdStatus.shouldCheckMessage());
    }

    @Test
    public void initFlagsを実行するとフラグは全てfalseを返却すること() {
        objCmdStatus.setOkMessageHit(true);
        objCmdStatus.setWarnMessageHit(true);
        objCmdStatus.setErrorMessageHit(true);
        objCmdStatus.resetFlags();
        assertFalse(objCmdStatus.isOkMessageHit());
        assertFalse(objCmdStatus.isWarnMessageHit());
        assertFalse(objCmdStatus.isErrorMessageHit());
    }

    @Test
    public void checkMessageLineで正常終了しましたを評価すると正常該当フラグだけがtrueとなること() {
        objCmdStatus.setOkMessageCsv("正常,いけてる,完璧");
        objCmdStatus.setWarnMessageCsv("警告,いけてるないかも,完璧でなかった");
        objCmdStatus.setErrorMessageCsv("異常,いけてない,失敗");
        objCmdStatus.initialize();
        objCmdStatus.resetFlags();
        objCmdStatus.checkMessageLine("正常終了しました");
        assertTrue(objCmdStatus.isOkMessageHit());
        assertFalse(objCmdStatus.isWarnMessageHit());
        assertFalse(objCmdStatus.isErrorMessageHit());
    }

    @Test
    public void checkMessageLineで警告が発生しましたがなんとか終了しましたを評価すると警告該当フラグだけがtrueとなること() {
        objCmdStatus.setOkMessageCsv("正常,いけてる,完璧");
        objCmdStatus.setWarnMessageCsv("警告,いけてるないかも,完璧でなかった");
        objCmdStatus.setErrorMessageCsv("異常,いけてない,失敗");
        objCmdStatus.initialize();
        objCmdStatus.resetFlags();
        objCmdStatus.checkMessageLine("警告が発生しましたがなんとか終了しました");
        assertFalse(objCmdStatus.isOkMessageHit());
        assertTrue(objCmdStatus.isWarnMessageHit());
        assertFalse(objCmdStatus.isErrorMessageHit());
    }

    @Test
    public void checkMessageLineで異常終了しましたを評価すると異常該当フラグだけがtrueとなること() {
        objCmdStatus.setOkMessageCsv("正常,いけてる,完璧");
        objCmdStatus.setWarnMessageCsv("警告,いけてるないかも,完璧でなかった");
        objCmdStatus.setErrorMessageCsv("異常,いけてない,失敗");
        objCmdStatus.initialize();
        objCmdStatus.resetFlags();
        objCmdStatus.checkMessageLine("異常終了しました");
        assertFalse(objCmdStatus.isOkMessageHit());
        assertFalse(objCmdStatus.isWarnMessageHit());
        assertTrue(objCmdStatus.isErrorMessageHit());
    }

    @Test
    public void checkMessageLineで警告が発生しましたが正常に終了しましたを評価すると正常と警告該当フラグがtrueとなること() {
        objCmdStatus.setOkMessageCsv("正常,いけてる,完璧");
        objCmdStatus.setWarnMessageCsv("警告,いけてるないかも,完璧でなかった");
        objCmdStatus.setErrorMessageCsv("異常,いけてない,失敗");
        objCmdStatus.initialize();
        objCmdStatus.resetFlags();
        objCmdStatus.checkMessageLine("警告が発生しましたが正常に終了しました");
        assertTrue(objCmdStatus.isOkMessageHit());
        assertTrue(objCmdStatus.isWarnMessageHit());
        assertFalse(objCmdStatus.isErrorMessageHit());
    }

    @Test
    public void isAlwaysNormalが正の場合はCheckCmdExitCodeの評価で結果がLVL_Iになること() {
        objCmdStatus.setAlwaysNormal(true);
        objCmdStatus.checkCommandExitCode(9999);
        assertEquals(MdlConst.LVL_I, objCmdStatus.getMethodExitStatus());
        assertEquals(MdlConst.LVL_I, objCmdStatus.getReturnLevel());
    }

    @Test
    public void 警告閾値と異常閾値が設定されていない場合にCheckCmdExitCodeの評価で0の結果がLVL_Iになること() {
        int intCmdRetcode = 0;
        objCmdStatus.setWarnThreshold(MdlConst.INT_NULL);
        objCmdStatus.setErrorThreshold(MdlConst.INT_NULL);
        objCmdStatus.checkCommandExitCode(intCmdRetcode);
        assertEquals(MdlConst.LVL_I, objCmdStatus.getMethodExitStatus());
        assertEquals(MdlConst.LVL_I, objCmdStatus.getReturnLevel());
    }

    @Test
    public void 警告閾値と異常閾値が設定されていない場合にCheckCmdExitCodeの評価で1の結果レベルがLVL_Eになること() {
        int intCmdRetcode = 1;
        objCmdStatus.setWarnThreshold(MdlConst.INT_NULL);
        objCmdStatus.setErrorThreshold(MdlConst.INT_NULL);
        objCmdStatus.checkCommandExitCode(intCmdRetcode);
        assertEquals(intCmdRetcode, objCmdStatus.getMethodExitStatus());
        assertEquals(MdlConst.LVL_E, objCmdStatus.getReturnLevel());
    }

    @Test
    public void 警告閾値と異常閾値が設定されていない場合にCheckCmdExitCodeの評価で異常時終了コードが20の場合1の戻り値は20になること() {
        int intCmdRetcode = 1;
        objCmdStatus.setWarnThreshold(MdlConst.INT_NULL);
        objCmdStatus.setErrorThreshold(MdlConst.INT_NULL);
        objCmdStatus.setErrorCode(20);
        objCmdStatus.checkCommandExitCode(intCmdRetcode);
        assertEquals(20, objCmdStatus.getMethodExitStatus());
        assertEquals(MdlConst.LVL_E, objCmdStatus.getReturnLevel());
    }

    @Test
    public void 新設短縮メソッド_getErrRetCodeCsvおよびisErrAtNegativeが正しく動作すること() {
        objCmdStatus.setErrRetCodeCsv("1,2,3");
        assertEquals("1,2,3", objCmdStatus.getErrRetCodeCsv());

        objCmdStatus.setErrAtNegative(true);
        assertTrue(objCmdStatus.isErrAtNegative());

        objCmdStatus.setWarnRetCodeList(java.util.List.of(10, 20));
        assertEquals(2, objCmdStatus.getWarnRetCodeList().size());

        objCmdStatus.setErrRetCodeList(java.util.List.of(30, 40));
        assertEquals(2, objCmdStatus.getErrRetCodeList().size());
    }
}
